### DISCUSSION TITLE
FUN active uptake terminology

### DISCUSSION THREAD

**Discussion by ecaas** [ESCOMP/CTSM#2098] [Q&A]
I'm currently looking into variables related to the FUN code, and I'm a bit confused about the terminology between mycorrhizal and nomycorrhizal N uptake, and cost related to it.

In the code, it seems like the nonmycorrhizal pathway counts as "active uptake":
https://github.com/ESCOMP/CTSM/blob/f47ce5cf1ab0223670e13f2ebccd876ec98ec09d/src/biogeochem/CNFUNMod.F90#L1512 
https://github.com/ESCOMP/CTSM/blob/f47ce5cf1ab0223670e13f2ebccd876ec98ec09d/src/biogeochem/CNFUNMod.F90#L1496

while in the declarations, and in the history field longnames, these variables are described as mycorrhizal only: 
https://github.com/ESCOMP/CTSM/blob/f47ce5cf1ab0223670e13f2ebccd876ec98ec09d/src/biogeochem/CNFUNMod.F90#L586-L587
https://github.com/ESCOMP/CTSM/blob/f47ce5cf1ab0223670e13f2ebccd876ec98ec09d/src/biogeochem/CNFUNMod.F90#L632-L633

In the documentation the nonmycorrhizal pathway is listed under the header "Cost of Active uptake" (https://escomp.github.io/ctsm-docs/versions/release-clm5.0/html/tech_note/FUN/CLM50_Tech_Note_FUN.html#cost-of-active-uptake), however only the mycorrhizal cost is labeled as "active".  

Is the nonmycorrhiza pathway in fact active, and the longnames and descriptions should be updated, or am I missing something?

The background for the issue is that in my single-site simulations of Norwegian forests, I find that the active N uptake is completely dominated by the nonmycorrhizal pathway, as shown in the below figure. Based on [Shi et. al 2016](https://onlinelibrary.wiley.com/doi/full/10.1111/gcb.13131#pane-pcw-figures), I would expect a dominance of the ECM pathway, and therefore wonder if I am interpreting the variables correctly.

![image](https://github.com/ESCOMP/CTSM/assets/45205620/f3c0ef72-a907-4259-abba-4db2b5f556a9)

**Comment by wwieder** [comment]
These are good questions.

It looks like the variable `n_active_*_acc` is used to sum up different pathways for active N uptake, with the ecm diagnostic saved here.

https://github.com/ESCOMP/CTSM/blob/f47ce5cf1ab0223670e13f2ebccd876ec98ec09d/src/biogeochem/CNFUNMod.F90#L1418

To circle back to your question about large non-mycorrhizal fluxes, I've seen similar patterns in global-scale runs.
If you look at this [table from CLM5 results](https://webext.cgd.ucar.edu/I20TR/clm50_r270_1deg_GSWP3V1_isofix_newpopd_hist/lnd/clm50_r270_1deg_GSWP3V1_isofix_newpopd_hist.1995_2014-clm50_clm50d001_1deg_CRUNCEPV7_isofix_hist.1995_2014/set5/set5_cn.txt):

- Total Nuptake ~ 1000 Tg N y-1
- AM N uptake ~ 100 Tg N y-1
- ECM N uptake ~ 10 Tg N y-1

We haven't really worried about the different active N uptake pathways being used in FUN, because it doesn't meaningfully feedback on the soil biogeochemistry. With your changes with MIMICS, however, these feedbacks matter.  I'd suggest modifying the FUN costs associated with different uptake pathways so that you generate fluxes that seem more appropriate for the system you're working in (e.g., higher active N uptake costs, or lower ECM N uptake costs, should favor the ECM pathway).  Just check that vegetation doesn't die out when you run the model from a cold start with the new parameter values.

Hopefully this helps.

**Reply by ecaas** [reply]
Thanks Will, this was helpful! 
I'll try to modify the FUN costs in the way you suggested to allocate more carbon the the ECM pathway.

**Reply by ecaas** [reply]
Hi again! 

I started looking into the costs associated with the different pathways, and think I might have found the cause of the high fraction going the non-mycorrhizal pathway. I tried to replicate [Fig 3](https://agupubs.onlinelibrary.wiley.com/cms/asset/b1baa6b8-c6c7-4d3d-a584-e94f622a9f27/jgrg20269-fig-0003-m.jpg) in [Brzostek et al. 2014](https://agupubs.onlinelibrary.wiley.com/cms/asset/b1baa6b8-c6c7-4d3d-a584-e94f622a9f27/jgrg20269-fig-0003-m.jpg) using parameters from the parameterfile ctsm51_params.c211112.nc (for needleleaf boreal evergreen tree). 

I suspect that the values for "kc_nonmyc" and "kn_nonmyc" in the parameterfile have been switched, because when I use the parameters as they are, the cost of nonmyc becomes very low (a and b in figure below), especially for the constant N case (b). If I switch the values I get plots more in line with what was shown in Brzostek et al. The pattern for the mycorrhizal pathways seems reasonable, so I will try to run a test case using switched values for "kc_nonmyc" and "kn_nonmyc". If you agree that this might be a bug, should I create a separate issue for that?

![figure_costs](https://github.com/ESCOMP/CTSM/assets/45205620/c406bf81-6af5-457f-bb71-925ef0012106)

**Reply by rosiealice** [reply]
Hi elin,

Apologies in my delay getting to this. And thanks for figuring it all out! I think you might be right on the switching, though it would tbh take me a while to get my head back into this space sufficiently to know the answer! Do you want to meet up some time wp ext week to discuss it? We can arrange offline...

**Reply by wwieder** [reply]
Any updates on this?  Just to clarify is CLM5 using the top row of plots, but we should be using the parameters from the bottom row of plots?  If this is accurate, creating a bug report would be helpful.  Then we can evaluate the impacts this on simulations in subsequent tests.

**Reply by rosiealice** [reply]
Elin and Terje and Betty and I met yesterday to talk this through. It does seem like the parameters should be switched to enable more C allocation to mycorhizzae. As you said Will, this didn't matter when we lumped all this together as 'active' but for MIMICS+ it of course affects how much fungi it can build! While the provenance of the parameter swap is a bit hazy, it seems to have been present at least as far back as the table in the Brzostek paper...

**Reply by ecaas** [reply]
Hi Will, 

To answer your question from above: Yes, CLM uses the values from the top two panels, while the bottom panels give costs more in line with the figure of the costs in Brzostek et al. 

I've also tried to create the plot using the values from Table S2 that you posted below, and need to do the same switch of parameters (kc << kn) instead of (kc >> kn) for the nonmycorrhizal uptake to get the same high cost as plotted in the figure. 

I did quick test run for a single site using switched parameters, and it seems to have minor effects on total carbon, NPP etc. but have a big impact on which pathway uptake that is preferred (now most uptake uses the EcM pathway instead of non-mycorrhizal). I'll file a bug report and include more details there.

**Reply by rosiealice** [reply]
I mentioned this issue to Renato and Josh,  and Renato pointed me to a paper which updates these numbers of which I wasn't aware...

https://www.frontiersin.org/articles/10.3389/ffgc.2020.00043/full

So we might be good looking tat how those compare too, even though, as I mentioned to Elin, the numbers might not compare directly anymore as CLMs layered model needs to use concentrations and not column total N... 

Renato also mentioned he has global maps of Kc and Kn we could think about too.

**Reply by ecaas** [reply]
Thanks, Rosie! 
I also read this paper https://onlinelibrary.wiley.com/doi/abs/10.1111/ele.12802 that uses a kc >> kn, so I'm not so sure about my suggested solution anymore. A global map would be very helpful for getting a better understanding of the parameters:) 

Anyway, I just filed a bug report #2120 and we'll see where it leads!

**Comment by wwieder** [comment]
OK, for what it's worth, I'm pasting the FUN2.0 parameters from Brzostek et al. (Table S2) and Fun 3.0 from Allen et al (SI Table 1) below:

I'm not sure how this compares the values used for ecto and AM fungi, but the active uptake values seem close to the same with Kc>>Kn in FUN2.0, whereas they are equal in FUN3.0

Cost Parameter | FUN 2.0 | FUN 3.0
-- |  -- | --
KC | 0.6 | 0.15
KN |  0.01 | 0.15
KR |  0.08 | 0.025
EKC |  0.3 | 0.15 
EKN | 0.05 | 0.025
AKC |  0.05 | 0.025
AKN |  0.1 | 0




</body>

</html>
